### PR TITLE
Move docker image tag logic to utils.groovy

### PR DESCRIPTION
**Pull request by cm421**
### Intent

Moves docker tag logic into `utils.groovy`, as suggested by https://github.com/rstudio/rstudio-pro/pull/4697/files/86533e8e3ec70c09476a92ae51c02fe8a12cd8ad#r1207141524

### Approach

> Describe the approach taken and the tradeoffs involved if non-obvious; add an overview of the solution if it's complicated.

### Automated Tests

> Indicate whether this change includes unit tests or integration tests, or link a work item or pull request to add those tests to another repo. If the change cannot or will not be covered by a test, indicate why.

### QA Notes

> Add additional information for QA on how to validate the change, paying special attention to the level of risk, adjacent areas that could be affected by the change, and any important contextual information not present in the linked issues. 

### Checklist

- [ ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [ ] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [ ] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [ ] This PR passes all local unit tests

### CODE REVIEW

File: utils.groovy [hunk 1]
```diff
@@ -213,4 +213,8 @@ def getDockerBuildOs(String osName) {
   }
 }
 
+def getDockerTag() {
+  return "${env.IS_PRO ? 'pro-' : ''}${env.OS}-${env.ARCH}-${env.RSTUDIO_VERSION_FLOWER}"
```
Comment (MariaSemple) [inline utils.groovy:217]: I think we still need to call `getDockerBuildOs` instead of passing `env.OS` directly

Comment (cm421) [reply utils.groovy]: Good catch!

**Review by AndrewMcClain — CHANGES_REQUESTED** [review]
Could you modify `Jenkinsfile.linux` to also use this tagging, that way we're consistent across builds.

**Comment by cm421** [comment]
Added `Jenkinsfile.pull-request` and `Jenkinsfile.windows` as well

**Review by AndrewMcClain — APPROVED** [review]
Excellent catches on the windows jenkinsfile as well!

LGTM!

**Review by MariaSemple — APPROVED** [review]
LGTM!

**Pull request merged**